aura/lib/Aura/Settings/External.hs

```diff
@@ -45,17 +46,28 @@ data AuraConfig = AuraConfig
   , acAnalyse   :: !(Maybe BuildSwitch) }
   deriving stock (Show)
 
-defaultAuraConf :: FilePath
-defaultAuraConf = "/etc/aura.conf"
+userAuraConfPath :: IO FilePath
+userAuraConfPath = getUserConfigFile "aura" "config"
 
-getAuraConf :: FilePath -> IO Config
-getAuraConf fp = do
-  exists <- doesFileExist fp
+systemAuraConfPath :: FilePath
+systemAuraConfPath = "/etc/aura.conf"
+
+-- | Attempt to get a valid Aura config from a specified path.
+getAuraConfFrom :: FilePath -> IO (Maybe Config)
+getAuraConfFrom path = do
+  exists <- doesFileExist path
   if not exists
-    then pure $ Config mempty
+    then pure Nothing
     else do
-      file <- decodeUtf8Lenient <$> BS.readFile fp
-      pure . either (const $ Config M.empty) id $ parse config "aura config" file
+      file <- decodeUtf8Lenient <$> BS.readFile path
+      pure . hush $ parse config "aura config" file
+
+getAuraConf :: IO Config
+getAuraConf = do
+  userCfg <- getAuraConfFrom =<< userAuraConfPath
+  case userCfg of
+    (Just x) -> pure x
```

Comment:
Parens are unnecessary here. Have you checked out [hlint](https://github.com/ndmitchell/hlint) yet?

Reply:
I have. It works on my other Haskell code but everything in this repo gives me back "No hints" for some reason... something I need to investigate.